Updated Salesforce docs regarding Invoke Apex Rest (#408)
mulesoft/docs-connectors: modules/ROOT/pages/salesforce/salesforce-connector-config-topics.adoc

Every time a subscription to `/u/TestStreaming` receives an event, it executes the rest of the flow. In the case of this example, it logs a message at INFO level.

> 

Reviewer: lines 186 and 345 - 389.

> NOTE: The *Streaming Channel* field of the Subscribe Streaming Channel operation does not display change events that are available in the Salesforce environment because this information is not provided in the Salesforce API. However, your connector can subscribe to a streaming channel. For example, to subscribe to the `All Change Events` channel, use `/data/ChangeEvents` as the channel name to which you subscribe. See https://developer.salesforce.com/docs/atlas.en-us.change_data_capture.meta/change_data_capture/cdc_subscribe_channels.htm[Subscription Channels] in the Salesforce Change Data Capture Developer Guide.

Reviewer: This is pretty long for a Note; do you suppose it should be part of the running narrative?

Also, I wonder about the placement of this information. This topic is titled Subscribe to a Streaming Channel, so is there a reason to say, after the rest of the topic text, that you can solve this problem by doing the very thing that the topic is about? Isn't this issue "covered" in the previous text?

These questions are for your consideration only; you need not respond to me directly.

One last question: would this note be just as effective if it comprised only the first and last sentences: the issue and the cross-ref?

You can decide which fields to set to Null without being forced to use the `fieldsToNull` field.

=== Upsert

Unless you configure the External ID Field Name for the sObject to which you are trying to upsert, every use of the upsert fails.

The upsert operation does not work with the sObject `priceBookentry2`.

While you can't change the `contentType` for bulk upsert, you can use the `Create Job` operation to set the contentType to either CSV or zipped CSV (if you're near the character limit). Follow up with the `Create Batch` operation.

=== Query

Even though you can see the fields of an sObject and their corresponding types via DataSense, the `Query` operation returns all fields as `String`.

If you want to use the actual type of the field, you must convert that field to the desired type using a Transform (or Transform Message) component.

Although `CreatedDate` field appears as `dateTime`, the query returns a `String` representing the date.

To actually use the field as a `dateTime`, you can configure it using Transform Message.

To store `Date` and `dateTime` fields, you can use DataWeave expressions to create `Date` and `Calendar` Java objects.

=== Invoke APEX Rest Method

> The Invoke APEX Rest operation enables users to invoke a method from an Apex class that is exposed as a REST service. The following example shows how a payload for this operation:

Reviewer: ```suggestion
The Invoke APEX Rest operation enables users to invoke a method from an Apex class that is exposed as a REST service. The following example shows a payload for this operation:
```

[source,example,linenums]
----
<ee:transform doc:name="Transform Message">
			<ee:message >
				<ee:set-payload ><![CDATA[output application/java
---
{
	body: {
		URLParameters: {
			Parameter1: "parameter1Value",
			Parameter2: "parameter2Value"
		},
		account: {
			Name: "Example",
			AccountNumber: "55"
		}
	},
	headers: {
		header1:"header1Value"
	},
	cookies: {
		cookie1:"cookie1Value"
	},
	queryParameters: {
		queryParam1Name:"queryParam1Value",
		queryParam2Name:"queryParam2Value"
	}
}]]></ee:set-payload>
			</ee:message>
		</ee:transform>
----

> The `body` contains `URLParameters`, which is a map containing the parameters that replace the wildcards in the path of the REST resource described in the Apex class. For example if the REST resource is set to `@RestResource(urlMapping='/myResource/\*/mySubResource/*')`, then the value of `Parameter1` replaces the first `\*` and the value of `Parameter2` replaces the second `*`. The key names must start with `Parameter` followed by a number showing the position of the `*` to be replaced. After the `URLParameters` block, provide the content of the `body` to be sent to the REST resource, as shown in the example `account` block.

Reviewer: ```suggestion
The `body` element contains `URLParameters`, which is a map containing the parameters that replace the wildcards in the path of the REST resource described in the Apex class. For example, if the REST resource is set to `@RestResource(urlMapping='/myResource/\*/mySubResource/*')`, then the value of `Parameter1` replaces the first `\*` and the value of `Parameter2` replaces the second `*`. The key names must start with `Parameter` followed by a number showing the position of the `*` to be replaced. After the `URLParameters` block, provide the content of the `body` to be sent to the REST resource, as shown in the example `account` block.
```

The `headers` and `cookies` fields describe the headers and cookies to pass along with the HTTP request to the desired service.

The `queryParameters` field describes the query parameters to use, and the keys and values in this map that have to be accepted by the specified Apex Class.

> === Insert Values in a Salesforce Drop-down

Reviewer: ```suggestion
=== Insert Values in a Salesforce Drop-Down Menu
```

> Inserting dependent values into an existing drop-down list field in Salesforce does not always work. Test to confirm functionality.

Reviewer: ```suggestion
Inserting dependent values into an existing drop-down menu in Salesforce does not always work. Test to confirm functionality.
```

Reviewer: Do we have any suggestions about how to test, or a link to such info?
